compiler/codegen/cg-expr.cpp

```diff
@@ -585,6 +585,9 @@ llvm::StoreInst* codegenStoreLLVM(GenRet val,
   // e.g. T3 = alloca i8;
   //      T3 = (T == T2);   // not actual LLVM syntax
   // in LLVM, boolean type is i1
+  if (val.val == NULL) {
```

Comment:
It seems odd to me that this gets all the way to codegen and then fails. (Of course that can be where some errors are caught, but it seems like something we might do better to detect earlier). What happens with the C backend?

Reply:
> It seems odd to me that this gets all the way to codegen and then fails.

I agree, and that's why I called it unsatisfying above. 

Great point about the fact that we're in LLVM-specific code here, though, which obviously makes my test case problematic...  The behavior is that we make it through to C compiler, but then blow up in an equivalent way, I think:

```
In file included from /path/to/_main.c:52:
/path/to/errorOnRuntimeTypeField.c:656:21: error: 
      unexpected type name
      '_array_DefaultRectangularArr_1_int64_t_F__real64_int64_t_chpl': expected
      expression
chpl_macro_tmp_33 = _array_DefaultRectangularArr_1_int64_t_F__real64_int...
                    ^
1 error generated.
```

So that definitely means we should move the check forward.  Just have to decide where to move it to (probably instantiation of default initializers?) and what to use to trigger the error (potentially reliance on a runtime type, though I currently haven't invested the time to understand why other related issues seem to fail in different ways... i.e., are there some cases that work OK today?).